src/Barcoder/DataMatrix/DataMatrixEncoder.cs

```diff
@@ -6,12 +6,23 @@ namespace Barcoder.DataMatrix
 {
     public static class DataMatrixEncoder
     {
-        public static IBarcode Encode(string content)
+        public static IBarcode Encode(string content, byte codeSize = 0)
         {
             var data = EncodeText(content);
 
-            CodeSize size = CodeSizes.All.FirstOrDefault(x => x.DataCodewords >= data.Length)
-                ?? throw new InvalidOperationException("Too much data to encode");
+            CodeSize size;
```

Comment:
This logic is getting a little bit too big for this `Encode` method and asks to be split off.

Methods could be:

```
private static CodeSize GetSmallestCodeSizeForData(int dataLength)
{
    return CodeSizes.All.FirstOrDefault(x => x.DataCodewords >= dataLength)
        ?? throw new InvalidOperationException($"No code size found that fits {dataLength} codewords");
}

private static CodeSize GetFixedCodeSizeForData(int fixedNumberOfRows, int dataLength)
{
    CodeSize codeSize = CodeSizes.All.FirstOrDefault(x => x.Rows == fixedNumberOfRows)
        ?? throw new InvalidOperationException($"No code size found with fixed number of rows {fixedNumberOfRows}");
    if (size.DataCodewords < dataLength)
        throw new ArgumentException($"The fixed code size does not fit {dataLength} codewords");
    return codeSize;
}
```

Then in `Encode`:

```
CodeSize size = fixedNumberOfRows.HasValue
    ? GetFixedCodeSizeForData(fixedNumberOfRows.Value, data.Length)
    : GetSmallestCodeSizeForData(data.Length);
```